TechDraw/TechDrawTools.FCMacro

```diff
@@ -70,6 +79,147 @@ class LineAttributes:
     def getWidth(self):
         return self.LineWidth
 
+class translate:
+    '''do all translations'''
+    def __init__(self):
+        self.langKey = self.getLanguage()
+
+    def getLanguage(self):
+        '''get local language key'''
+        langDict = {}
+        langDict["English"] = 0
+        langDict["German"] = 1
+        langDict["Italian"] = 2
+        langDict["French"] = 3
+        langDict["Spanish"] = 4
+        self.langKey = 0
+        if App.Gui.getLocale() in langDict.keys():
+            self.langKey = langDict[App.Gui.getLocale()]
+        return self.langKey
+
+    def tr(self,textEnglish):
+        if self.langKey == 0:
+            return self.englishDict(textEnglish)
+        elif self.langKey == 1:
+            return self.germanDict(textEnglish)
+        elif self.langKey == 2:
+            return self.italianDict(textEnglish)
+        elif self.langKey == 3:
+            return self.frenchDict(textEnglish)
+        elif self.langKey == 4:
+            return self.spanishDict(textEnglish)
+
+    def englishDict(self,textEnglish):
+        return textEnglish
+    
+    def germanDict(self,textEnglish):
+        dict = {"Attributes and Modifications":"Eigenschaften und Änderungen",\
+                "Centerlines and Threading":"Mittellinien und Gewinde",\
+                "Dimensions":"Bemaßungen",\
+                "Line attributes:":"Eigenschaften:",\
+                "Cascade spacing:":"Abstand Kaskadierung",\
+                "Quit":"Fertig",\
+                "set lines to ":"setze Linie auf ",\
+                "solid":"Vollinie",\
+                "dashed":"strichliert",\
+                "dotted":"punktiert",\
+                "dashdot":"strich-punkt",\
+                "set lines width to ":"setze Linienbreite auf ",\
+                "thick 0,70":"breit 0,70",\
+                "middle 0,35":"mittel 0,35",\
+                "thin 0,18":"schmal 0,18",\
+                "set lines color to ":"setze Linienfarbe auf ",\
+                "black":"schwarz",\
+                "grey":"grau",\
+                "red":"rot",\
+                "green":"grün",\
+                "blue":"blau",\
+                "magenta":"magenta",\
+                "cyan":"cyan",\
+                "yellow":"gelb"}
+        return dict[textEnglish]
```

Comment:
This will fail if `textEnglish` is not in the dictionnary, please use `dict.get(textEnglish, textEnglish)`.